sormas-api/src/main/java/de/symeda/sormas/api/user/UserRole.java

```diff
@@ -238,105 +223,52 @@ public void addAssignableRoles(Collection<UserRole> collection) {
 		return result;
 	}
 
-	public Collection<UserRole> getCombinableRoles() {
-
-		switch (this) {
-		case ADMIN:
-			return Arrays.asList(
-				SURVEILLANCE_SUPERVISOR,
-				CASE_SUPERVISOR,
-				CONTACT_SUPERVISOR,
-				EVENT_OFFICER,
-				LAB_USER,
-				NATIONAL_USER,
-				NATIONAL_OBSERVER,
-				NATIONAL_CLINICIAN,
-				IMPORT_USER);
-		case NATIONAL_USER:
-			return Arrays.asList(LAB_USER, ADMIN, NATIONAL_CLINICIAN, IMPORT_USER);
-		case NATIONAL_OBSERVER:
-			return Arrays.asList(ADMIN, IMPORT_USER);
-		case NATIONAL_CLINICIAN:
-			return Arrays.asList(ADMIN, NATIONAL_USER, IMPORT_USER);
-		case CASE_SUPERVISOR:
-		case CONTACT_SUPERVISOR:
-		case SURVEILLANCE_SUPERVISOR:
-		case EVENT_OFFICER:
-			return Arrays.asList(SURVEILLANCE_SUPERVISOR, CASE_SUPERVISOR, CONTACT_SUPERVISOR, EVENT_OFFICER, LAB_USER, ADMIN, IMPORT_USER);
-		case LAB_USER:
-			return Arrays
-				.asList(SURVEILLANCE_SUPERVISOR, CASE_SUPERVISOR, CONTACT_SUPERVISOR, EVENT_OFFICER, LAB_USER, NATIONAL_USER, ADMIN, IMPORT_USER);
-		case SURVEILLANCE_OFFICER:
-		case CASE_OFFICER:
-		case CONTACT_OFFICER:
-			return Arrays.asList(SURVEILLANCE_OFFICER, CASE_OFFICER, CONTACT_OFFICER, IMPORT_USER);
-		case HOSPITAL_INFORMANT:
-			return Arrays.asList(HOSPITAL_INFORMANT, IMPORT_USER);
-		case COMMUNITY_INFORMANT:
-			return Arrays.asList(COMMUNITY_INFORMANT, IMPORT_USER);
-		case EXTERNAL_LAB_USER:
-			return Arrays.asList(EXTERNAL_LAB_USER, IMPORT_USER);
-		case STATE_OBSERVER:
-		case DISTRICT_OBSERVER:
-		case POE_INFORMANT:
-		case POE_SUPERVISOR:
-		case POE_NATIONAL_USER:
-			return Arrays.asList(IMPORT_USER);
-		case IMPORT_USER:
-			final List<UserRole> userRoles = new ArrayList<>();
-			for (UserRole userRole : UserRole.values()) {
-				if (userRole != REST_EXTERNAL_VISITS_USER && userRole != REST_USER) {
-					userRoles.add(userRole);
-				}
-			}
-			return userRoles;
-		case REST_EXTERNAL_VISITS_USER:
-			return Arrays.asList(REST_EXTERNAL_VISITS_USER);
-		case REST_USER:
-			return Arrays.asList(REST_USER);
-		default:
-			throw new UnsupportedOperationException("getCombinableRoles not implemented for user role: " + this);
-		}
+	public JurisdictionLevel getJurisdictionLevel() {
+		return jurisdictionLevel;
 	}
 
-	public static boolean isSupervisor(Collection<UserRole> roles) {
+	/**
+	 * Expects the roles have been validated.
+	 * 
+	 * @param roles
+	 * @return
+	 */
+	public static JurisdictionLevel getJurisdictionLevel(Collection<UserRole> roles) {
 
 		for (UserRole role : roles) {
-			if (role.isSupervisor()) {
-				return true;
+			JurisdictionLevel jurisdictionLevel = role.getJurisdictionLevel();
+			if (jurisdictionLevel != JurisdictionLevel.NONE && jurisdictionLevel != JurisdictionLevel.LABORATORY) {
```

Comment:
I don't quite get why this method returns NONE if the jurisdiction level is LABORATORY. Can you explain that to me?

Reply:
it is a mistake or maybe a shortcoming of not having what we need, was added because laboratory is combinable with other roles
for LAB_USER -> should return LABORATORY jurisdiction - this is clear and will change it
for {LAB_USER, NATIONAL_USER} -> it should return NATION - is clear and works like this
for {LAB_USER,ADMIN} -> it should return NONE or LABORATORY jurisdiction?

Reply:
I'm not exactly sure; since you've created the logic, maybe you can answer the following question:

* If LABORATORY is used, will the admin part of the user have any problems? 
* If NONE is used, will the lab user part of the user have any problems?

If the answers to both questions are "no", then I'd probably say that having a jurisdiction should have priority over having no jurisdiction. What do you say?